docs/src/idioms/option-result.md

```diff
@@ -0,0 +1,12 @@
+# Options and Nullables
+
+In keeping with uniffi guidelines, we have made an effort to map core Rust concepts into their Typescript equivalents wherever possible, in order to allow the Typescript code to be as idiomatic and easy to work with as possible. A returned `Result<T, E>` from Rust will result in a flat type of `T` or a thrown `E`, while an `Option<T>` will become `T | undefined`. We believed that flattened types like this would be easier to work with than wrappers at every point for such common Rust primitives. This in in keeping with the approach taken by the core uniffi team for other languages (Kotlin turns `Option<T>` into `T`?).
```

Comment:
```suggestion
In keeping with uniffi guidelines, we have made an effort to map core Rust concepts into their Typescript equivalents wherever possible, in order to allow the Typescript code to be as idiomatic and easy to work with as possible. A returned `Result<T, E>` from Rust will result in a flat type of `T` or a thrown `E`, while an `Option<T>` will become `T | undefined`. We believed that flattened types like this would be easier to work with than wrappers at every point for such common Rust primitives. This is in keeping with the approach taken by the core uniffi team for other languages (Kotlin turns `Option<T>` into `T`?).
```